include/particles/particle.h

```diff
@@ -117,26 +120,17 @@ class Particle : public ParticleBase<Tdim> {
   bool assign_volume(double volume) override;
 
   //! Return volume
-  double volume() const override { return volume_; }
+  double volume() const override {
+    return this->scalar_property(mpm::properties::Scalar::Volume);
+  }
```

Comment:
The current version define all simple return value functions in the derive class. How can I define such function for two phase? Such as `liquid_mass()` `porosity()` `liquid_material()`

Reply:
If it is inside your `twophase_particle`, you can create a private function to ease the calling process, for instance:
```
private:
double pore_pressure() const override {
    return this->scalar_property(mpm::properties::Scalar::PorePressure);
}
```
However, to call it from the `twophase_particle_functions`, you can use:
```
particle->scalar_property(mpm::properties::Scalar::PorePressure);
```

Reply:
Yes, I need it in the `twophase_particle_functions`. I can do this way, but it will be not identical to the single phase format.

Reply:
I understand. That should be ok.

Reply:
If you use the variable multiple times, maybe you want to define it in the beginning of the function, such that:
`const auto pore_pressure = particle->scalar_property(mpm::properties::Scalar::PorePressure);`
and you can use it multiple times.

Reply:
This also means a particle pointer should be set as an argument of the functions.

Reply:
Yes, you are right, the same as other functions in `particle_functions`.